src/DenseACSets.jl

```diff
@@ -643,6 +702,64 @@ function replace_colons(acs::ACSet, parts::NamedTuple{types}) where {types}
     part == (:) ? (1:nparts(acs, type)) : part
   end)
 end
+# Garbage collection 
+####################
+"""
+Returns the part type of an arbitrary part of an ACSet (we presently assume 
+that all parts are the same type). If there are no objects in the schema, we 
+return nothing
+"""
+function part_type(X::ACSet)
+  S = acset_schema(X)
+  if !isempty(ob(S))
+    p = X.parts[first(ob(S))]
+    if typeof(p) == IntParts
+      return Int 
+    elseif typeof(p) == BitSetParts
+      return BitSet
+    else 
+      error("Unknown part type $(typeof(p))")
+    end
+  end
+end
+
+"""
+Reindex the parts of the acset such that there are no gaps between the indices.
+Return a vector for each part mapping the new parts into the old parts. 
+"""
+function ACSetInterface.gc!(X::ACSet)
+  if part_type(X) == Int 
+    return Dict(o=>1:nparts(X,o) for o in types(X))
+  elseif part_type(X) == BitSet
+    S = acset_schema(X)
+    μ = Dict(map(types(S)) do o 
+      p    = X.parts[o]
+      m    = collect(p.val)
+      m⁻¹  = [findfirst(==(i), m) for i in 1:p.next.x]
+      return o => (m,m⁻¹)
+    end)
+    # Update homs and attrs
+    for (h, a, b) in homs(S)
+      μᵦ = μ[b][2]
+      X[h] = [μᵦ[X[a,h]] for a in μ[a][1]]
+      for i in (nparts(X,a)+1) : X.parts[a].next.x
+        clear_subpart!(X, i, h)
+      end
+    end
```

Comment:
maybe throw in a `sizehint!` at the end here? Can't remember if columns support that or not.